src/auth_storage.js

```diff
@@ -0,0 +1,98 @@
+export class LocalStorage {
+  constructor (prefix = 'cozy:oauth:', getLocalStorage = () => window.localStorage) {
+    // We try to deref the localstorage, and if it fails (because the user has
+    // deactivated its localstorage for instance) we fallback on a memory
+    // storage.
+    let storage
+    try {
+      storage = getLocalStorage()
+    } catch (e) {
+      storage = memStorage
+    }
+    this.storage = storage
+    this.prefix = prefix
+  }
+
+  save (key, value) {
+    return new Promise(resolve => resolve(
+      this.storage.setItem(this.prefix + key, JSON.stringify(value))))
+  }
+
+  load (key) {
+    return new Promise(resolve => {
+      const item = this.storage.getItem(this.prefix + key)
+      if (!item) {
+        resolve()
+      } else {
+        resolve(JSON.parse(item))
+      }
+    })
+  }
+
+  delete (key) {
+    return new Promise(resolve => resolve(
+      this.storage.removeItem(this.prefix + key)))
+  }
+
+  clear () {
+    return new Promise(resolve => {
+      const storage = this.storage
+      for (let i = 0; i < storage.length; i++) {
+        const key = storage.key(i)
+        if (key.indexOf(this.prefix) === 0) {
+          storage.removeItem(key)
+        }
+      }
+      resolve()
+    })
+  }
+}
+
+export class MemoryStorage {
+  constructor () {
+    this.hash = {}
```

Comment:
You should probably use an ES2015 Map, or `Object.create(null)`, instead of just `{}`